lib/mapObjects/ObjectTemplate.cpp

```diff
@@ -287,7 +287,15 @@ void ObjectTemplate::readJson(const JsonNode &node, const bool withTerrain)
 	if(withTerrain && !node["allowedTerrains"].isNull())
 	{
 		for(auto& entry : node["allowedTerrains"].Vector())
-			allowedTerrains.insert(VLC->terrainTypeHandler->getInfoByName(entry.String())->id);
+		{
+			try {
+				allowedTerrains.insert(VLC->terrainTypeHandler->getInfoByName(entry.String())->id);
```

Comment:
getInfoByName returns pointer but throws exception instead of returning nullptr which doesn't make much sense
It's needed to be refactored in future in the way of either returning reference or returning nullptr without exception

Reply:
Yup. Problem is, there are like 5 or so usages of that method alone plus unknown number of usages of similar methods for roads/rivers. And haven't seen any obvious try/catch blocks around others.

Don't want to change all of them, so for now I'd settle on "Failed to find terrain %s for object %s" message which is much more useful for modders than current "exception thrown by map::at" or something like that

Reply:
Yeah, agree that for release branch no need to make such refactoring, just noting without action needed